finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
Ok, I believe now I finally understand what do you mean :). So you are suggesting to implement a server `SslConnectHandler` and delay `channelRegistered` in it? It should be possible, but we also will have to change `ServerBridge` to be a handler.